File: LeetCode/LeetCode Practice Problems/1299- Replace Elements with Greatest Element on Right Side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& a) {
        int maxi = -1;
        int n = a.size();
        for(int i=n-1;i>=0;i--)
        {
            int v = a[i];
            a[i] = maxi;
            maxi = max(v,maxi);
        }
        return a;
    }
};

File: LeetCode/LeetCode Practice Problems/1299-Replace Elements with Greatest Element on Right Side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int mx = -1;
        for(int i=arr.length-1;i>=0;i--) {
            int val = arr[i];
            arr[i] = mx;
            mx = Math.max(mx , val);
        }
        
        return arr;
    }
}
